plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php

```diff
@@ -340,6 +353,56 @@ function ( $subscription ) {
 		}
 	}
 
+	/**
+	 * Runs on in_plugin_update_message-{file-name}, show a message if plugin is without a subscription.
+	 * Only Woo local plugins are passed to this function.
+	 *
+	 * @see setup_message_for_plugins_without_subscription
+	 * @param object $plugin_data An array of plugin metadata.
+	 * @param object $response  An object of metadata about the available plugin update.
+	 *
+	 * @return void.
+	 */
+	public static function display_notice_for_plugins_without_subscription( $plugin_data, $response ) {
+		// Extract product ID from the response.
+		$product_id = preg_replace( '/[^0-9]/', '', $response->id );
+
+		wc_get_logger()->debug( 'Checking if plugin has subscription', array( 'product_id' => $product_id ) );
+
+		if ( WC_Helper::has_product_subscription( $product_id ) ) {
+			return;
+		}
+
+		wc_get_logger()->debug( 'Nope' );
```

Comment:
is this a test logger?

Reply:
Thank you, fixed in https://github.com/woocommerce/woocommerce/pull/51060/commits/a6c781bd25c3d0eac054cf8d2765d5cffaff570d